rfcs/0021-glossary-terms.md

```diff
@@ -0,0 +1,123 @@
+
+**Rationale:** This term is used by other protocols, clearly describes the role duties, and is unambiguous.
+
+**Downsides:** EOS is the most popular chain that uses "block producers" and there is a chance of confusion if people think Coda also has only 21 block producers - but this is a tail risk.
+
+**Usage:** A block producer or block producing node stakes coda to participate in consensus.
+
+**Alternatives:**
+- Validator - used by most other protocols, but not suitable for Coda, as every node can be a "verifier" or "validator" of SNARKs.
+- Staker - tied to a specific consensus model, and doesn't explain what the node does for the network. It also has an edit distance of 2 from snarker, which can cause some confusion.
+- Proposer - currently used in codebase, but unused by other protocols, and also sounds less firm / tied to generating blocks (relative to producer).
+- Prover - currently used to describe the action of generating the block SNARK, but if this that job moves to snark workers, then this term is irrelevant.
+- Miner, Baker, etc - not relevant to Coda's consensus mechanism.
+
+### Block
+**Concept:** a set of transactions that represent a state transition.
+
+**Rationale:** This term aligns with the rest of the industry, and it makes it easier to onboard other blockchain users.
+
+**Downsides:** Perhaps it is not as technically accurate as a transition.
+
+**Usage:** In the Coda network, transactions get added to blocks.
+
+**Alternatives:**
+- External transition - maybe this is more precise, but has the heavy lifting of having to explain a new term for a mostly similar concept. Additionally, the word transition is very similar to transaction, causing confusion at times.
+
+### User Transaction
+**Concept:** A transaction issued by a user - currently a payment or a delegation change
+
+**Rationale:** This term clearly describes the concept, and leaves room for further types of user issued transactions.
+
+**Downsides:** This is a subset of "Transactions", so it is a bit redundant.
+
+**Usage:** There are three types of transactions in the Coda network currently - user transactions, fee transfers, and coinbases.
+
+**Alternatives:**
+- User command - this term is confusing, especially in  docs, as it can be conflated with a CLI command that a user issues.
+
+### Snark worker / Snarker
+**Concept:** A Coda node that produces SNARKs to compress transactions (and in the future, blocks) in the network.
+
+**Rationale:** The community likes SNARKs and it is a point of differentiation for Coda, so snarker is an engaging term.
+
+**Downsides:** Gets invalidated if Coda switches to another type of zero-knowledge proof. NOTE - There is some concern about using the progressive tense of "snark" as a verb, eg. "snarking" - as this can be confused with staking. However, if the community enjoys "snarking", it  would make sense to  continue using it as a verb.
```

Comment:
The other downside is that the block producing process also involves creation of SNARKs

Reply:
The main alternative we are considering is STARKs, which IIUC, is just a type of SNARK, so I think it would be fine to still call them Snarkers. When you build a STARK proof, you are still "snarking".

Alternative name that could be more generic, if we still want it, would be "Prover".

Reply:
We can always switch to "starkers". And that change can be messaged as part of the "evolution" to newer tech, if we ever do it.

I think there is already a perception that STARKs are different and unique from SNARKs so keep "snarker" after a switch might be confusing to people - but honestly, I think we can cross this bridge when we get there, as it won't happen any time soon.